### ISSUE TITLE
Reimplement multiclass loss to make it more stable numerically

### ISSUE THREAD

**Issue by sumitpai** [Accenture/AmpliGraph#170]
### Description
Multiclass_loss = log (exp(score_pos)/sum(exp(score_pos) + exp(score_neg)))

when score_pos goes above certain value, the loss becomes inf/inf which results in NaN loss. 
Currently we can use high regularization to avoid this. However it sometimes does occur
